tableaudocumentapi/datasource.py

```diff
@@ -56,7 +59,11 @@ def __init__(self, dsxml, filename=None):
     @classmethod
     def from_file(cls, filename):
         "Initialize datasource from file (.tds)"
-        dsxml = ET.parse(filename).getroot()
+
+        if zipfile.is_zipfile(filename):
```

Comment:
Is this pattern repeated?  Seems like this could be a good candidate for a helper function.